src/executors/mod.rs

```diff
@@ -7,50 +7,47 @@ use std::collections::HashMap;
 use std::rc::Rc;
 use std::sync::Arc;
 
-use actix_web::{http::Method, web, HttpRequest};
-use anyhow::{bail, Result};
-use log::debug;
+use actix_web::HttpResponse;
+use anyhow::Result;
+use log::{debug, info};
+
 use pyo3_asyncio::TaskLocals;
 // pyO3 module
 use crate::types::PyFunction;
-use futures_util::stream::StreamExt;
+
 use pyo3::prelude::*;
 use pyo3::types::PyDict;
 
 /// @TODO make configurable
-const MAX_SIZE: usize = 10_000;
 
 pub async fn execute_middleware_function<'a>(
     function: PyFunction,
-    payload: &mut web::Payload,
+    payload: &mut [u8],
     headers: &HashMap<String, String>,
-    req: &HttpRequest,
     route_params: HashMap<String, String>,
     queries: Rc<RefCell<HashMap<String, String>>>,
     number_of_params: u8,
+    res: Option<&HttpResponse>,
 ) -> Result<HashMap<String, HashMap<String, String>>> {
     // TODO:
     // add body in middlewares too
 
-    let mut data: Vec<u8> = Vec::new();
-
-    if req.method() == Method::POST
-        || req.method() == Method::PUT
-        || req.method() == Method::PATCH
-        || req.method() == Method::DELETE
-    {
-        let mut body = web::BytesMut::new();
-        while let Some(chunk) = payload.next().await {
-            let chunk = chunk?;
-            // limit max size of in-memory payload
-            if (body.len() + chunk.len()) > MAX_SIZE {
-                bail!("Body content Overflow");
-            }
-            body.extend_from_slice(&chunk);
-        }
-
-        data = body.to_vec()
+    let data = payload.to_owned();
+    let tmp = &HttpResponse::Ok().finish();
```

Comment:
can you please rename to `temp_response` or something a bit more meaningful than `tmp`?

Reply:
done